src/containers/Tenant/Query/ResultControls/ResultControls.tsx

```diff
@@ -0,0 +1,129 @@
+}
+
+export function ResultControls<T extends string>({
+    error,
+    stats,
+    activeSection,
+    loading,
+    onSelectSection,
+    sectionOptions,
+    clipboardText,
+    isResultsCollapsed,
+    isFullscreenDisabled,
+    onCollapseResults,
+    onExpandResults,
+    onStopButtonClick,
+}: ResultControlsProps<T>) {
+    const [startTime, setStartTime] = React.useState<number>(0);
+    const [elapsedTime, setElapsedTime] = React.useState<number>(0);
+
+    const intervalRef = React.useRef<number>();
+
+    React.useEffect(() => {
+        if (loading) {
+            setStartTime(Date.now());
+        } else {
+            clearInterval(intervalRef.current);
+            setElapsedTime(0);
+        }
+    }, [loading]);
+
+    React.useEffect(() => {
+        intervalRef.current = window.setInterval(() => {
+            setElapsedTime(Math.floor((Date.now() - startTime) / SECOND_IN_MS));
+        }, SECOND_IN_MS);
+
+        return () => window.clearInterval(intervalRef.current);
+    }, [startTime]);
+
+    return (
+        <div className={b('controls')}>
+            <div className={b('controls-right')}>
+                <QueryExecutionStatus error={error} loading={loading} />
+
+                {!error && !loading && (
+                    <React.Fragment>
+                        {stats?.DurationUs !== undefined && (
+                            <QueryDuration duration={Number(stats.DurationUs)} />
+                        )}
+                        {sectionOptions && activeSection && onSelectSection && (
+                            <React.Fragment>
+                                <Divider />
+                                <RadioButton
+                                    options={sectionOptions}
+                                    value={activeSection}
+                                    onUpdate={onSelectSection}
+                                />
+                            </React.Fragment>
+                        )}
+                    </React.Fragment>
+                )}
+                {loading ? (
+                    <React.Fragment>
+                        <Label className={b('elapsed-label')}>
+                            {configuredNumeral(elapsedTime).format('00:00')}
+                        </Label>
+                        <Button onClick={onStopButtonClick} className={b('stop-button')}>
+                            <Icon data={StopFill} size={16} />
+                            {'Stop'}
+                        </Button>
+                    </React.Fragment>
+                ) : null}
+            </div>
+            <div className={b('controls-left')}>
+                {clipboardText !== undefined && (
+                    <ClipboardButton
+                        text={clipboardText}
+                        view="flat-secondary"
+                        title="Copy results"
+                    />
+                )}
+                <EnableFullscreenButton disabled={isFullscreenDisabled} />
```

Comment:
Fullscreen button doesn't work until result is ready

Reply:
got rid of ResultControls so now it works as it was before